examples/darkmode/src/components/ThemeToggleButton.astro

```diff
@@ -0,0 +1,97 @@
+---
+/* This button will switch between the default theme and the oppsite of that
+ *
+ * If the default theme is auto it will toggle between auto and the opposite of the systemTheme
+ *
+ * If the default theme is not auto it will never enter auto mode
+ */
+---
+
+<button is="theme-toggle-button"></button>
+<style>
+  button[is="theme-toggle-button"] {
+    background-color: transparent;
+    border: none;
+    color: inherit;
+    width: 2.5rem;
+    height: auto;
+    cursor: pointer;
+  }
+</style>
+<script is:inline>
+  if (!customElements.get("theme-toggle-button")) {
+    class ThemeToggleButton extends HTMLButtonElement {
+      constructor() {
+        super();
+      }
+
+      connectedCallback() {
+        this.render();
+        this.bindEvents();
+        this.updateAriaLabel();
+      }
+
+      render() {
+        this.innerHTML = `
+          <svg fill="none" viewBox="0 0 24 24" stroke="currentColor" id="moon" aria-hidden="true">
```

Comment:
How can people use an icon like from a react library?

Reply:
they're on their own! 

it's probably possible to make it easier though, but basically it has to be valid html so className etc will be a problem

Reply:
I guess they should resort to like `renderToString`